### ISSUE TITLE
vendor folder support?

### ISSUE THREAD

**Issue by mholt** [client9/misspell#108]
Thanks for this tool; I use misspell in Caddy's CI tests. We're vendoring our dependencies though and this is proving to be a problem for our checks because misspell does not work well with vendor folders, where I don't care if there are misspellings.

Old CI command:

```
$ misspell -error .
```

Works great, except it checks the `vendor/` folder, which I need to avoid.

```
$ misspell -error $(go list ./... | grep -v vendor)
```

Doesn't work because misspell doesn't take package paths.

```
$ find . -type f | grep -v vendor/ | xargs misspell
```

Checks more than just .go files.

```
find . -type d | grep -v vendor/ | xargs misspell
```

Same problem as first, since it still traverses into the vendor subfolder.

Basically, I'm torn. For now, I'm removing this from our CI checks, but if the first command could skip `vendor/` folders, that would be best for me I think and I could add this back to our checks.

**Comment by client9** [comment]
Biggest complaint I have is regarding file selection.  That's good since people like you want to use it!

I'm on it!

n

**Comment by mholt** [comment]
Thanks! I was pleased to find (just a few minutes ago) that somehow this tool works with `gometalinter --vendor` -- I'm not sure why that is, but maybe you'd know. Anyway, I think I can continue using it with gometalinter for now!

**Comment by client9** [comment]
Oh great, yeah I know since I added it to gometalinter :-)  Yeah it's a good tool for people using golang.  And that reminds me... I need to update it.

I'll still add file/path selection for those  who don't use go.

**Comment by client9** [comment]
i'll update the readme to highlight gometalinter too   thanks for writing in!

**Comment by client9** [comment]
FWIW @mholt I think this will work too:

```
 find . -type f -name '*.go' | grep -v vendor/ | xargs misspell -error
```

You run it again and look at `.txt`, `.md` and other files as well

```
 find . -type f -name '*.txt' | grep -v vendor/ | xargs misspell -error
```

There are no doubt crazy `find` and `grep` tricks that can do this all in one pass, this works and seems fast enough.

Also note to self: need to check out caddy more.  Great project!

**Comment by mholt** [comment]
@client9 Thanks for the tips! I'll look those up for next time, unix is amazing... and I do hope you find Caddy useful!
